lib/pages/caregiver/plan_details_page.dart

```diff
@@ -25,141 +26,118 @@ class CaregiverPlanDetailsPage extends StatefulWidget {
-							title: "Przygotuj książki i zeszyty na kolejny dzień według bardzo długiego planu zajęć",
-							timer: 60,
-							pointsValue: 100,
-							pointCurrency: UICurrency(type: CurrencyType.diamond, title: "Punkty")
-						)
-					)
-				),
-				Padding(
-					padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
-					child: TaskCard(
-						index: 2,
-						task: TaskFormModel(
-							key: ValueKey(DateTime.now()),
-							title: "Spakuj potrzebne rzeczy"
+				for (var task in tasks)
+					Padding(
+						padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
+						child: TaskCard(
+							index: tasks.indexOf(task),
+							task: TaskFormModel(
+								key: ValueKey(DateTime.now()),
+								title: task.name,
+								timer: task.timer,
+								pointsValue: task.points != null ? task.points.quantity : null,
+								pointCurrency:  task.points != null ? UICurrency(type: task.points.type, title: task.points.title) : null
+							)
 						)
 					)
-				),
-				Padding(
-					padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
-					child: TaskCard(
-						index: 3,
-						task: TaskFormModel(
-							key: ValueKey(DateTime.now()),
-							title: "Spakuj potrzebne rzeczy part 2",
-							timer: 20
-						)
-					)
-				)
 			]
 		);
 	}
 
-	Segment _getAdditionalTasksSegment({String title, String noElementsMessage}) {
+	Segment _getAdditionalTasksSegment({String title, String noElementsMessage, List<UITask> tasks}) {
 		return Segment(
 			title: title,
 			noElementsMessage: '$_pageKey.content.noTasks',
 			elements: <Widget>[
-				Padding(
-					padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
-					child: TaskCard(
-						task: TaskFormModel(
-							key: ValueKey(DateTime.now()),
-							title: "Opcjonalne zadanko",
-							timer: 20,
-							optional: true,
-							pointsValue: 300,
-							pointCurrency: UICurrency(type: CurrencyType.ruby, title: "Klejnoty")
+				for (var task in tasks)
+					Padding(
+						padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
+						child: TaskCard(
+							task: TaskFormModel(
+								key: ValueKey(DateTime.now()),
+								title: task.name,
+								timer: task.timer,
+								optional: true,
+								pointsValue: task.points != null ? task.points.quantity : null,
+								pointCurrency:  task.points != null ? UICurrency(type: task.points.type, title: task.points.title) : null
+							)
 						)
 					)
-				)
 			]
 		);
 	}
+
+	Widget _getCardHeader(UIPlan plan) {
+		return AppHeader.widget(
+			title: '$_pageKey.header.title',
+			appHeaderWidget: ItemCard(
+				title: plan.name,
+				subtitle: plan.description(context),
+				chips: <Widget>[
+					AttributeChip.withIcon(
+						content: AppLocales.of(context).translate('page.caregiverSection.plans.content.tasks', {'NUM_TASKS': plan.taskCount}),
+						color: Colors.indigo,
+						icon: Icons.layers
+					)
```

Comment:
What about the subject of putting assigned children names here instead of pretty redundant task number?